cpp/Marc21Tests.cc

```diff
@@ -0,0 +1,262 @@
+
+MarcRecord testMarc21(File * const input) {
+    MarcRecord record = MarcReader::Read(input);
+    if (not record) { return record; }
+
+    record.getControlNumber();
+    record.getFieldIndex("689");
+    record.extractFirstSubfield("689", 't');
+
+    std::vector<std::string> vector;
+    size_t count = record.extractAllSubfields("100", &vector, "a0");
+
+    count = record.extractSubfield("689", '0', &vector);
+
+    count = record.extractSubfields("689", "02", &vector);
+
+    std::vector<std::pair<size_t, size_t>> LOKs;
+    count = record.findAllLocalDataBlocks(&LOKs);
+
+    if (count > 0) {
+        std::vector <size_t> indizes;
+        count = record.findFieldsInLocalBlock("852", "?1", LOKs[0], &indizes);
+    }
+
+    record.filterTags({"LOK"});
+    count = record.findAllLocalDataBlocks(&LOKs);
+
+    Subfields subfields('x', 'y');
+    subfields.addSubfield('a', "Test");
+    ssize_t index = record.insertField("TST", subfields.toString());
+    record.extractFirstSubfield("TST", 'a');
+
+    Subfields newSubfields('a', 'b');
+    newSubfields.addSubfield('a', "FooBar");
+    record.updateField(index, newSubfields.toString());
+    record.extractFirstSubfield("TST", 'a');
+
+    record.deleteField(index);
+    record.getFieldIndex("TST");
+
+    for (size_t i = 0; i < INSERTED_FIELDS_COUNT; ++i) {
+        record.insertField("TST", subfields.toString());
+    }
+
+    return record;
+}
+
+MarcUtil::Record testOldRecord(File * const input) {
+    MarcUtil::Record record = MarcUtil::Record::BinaryFactory(input);
+    if (not record) { return record; }
+
+    record.getControlNumber();
+    record.getFieldIndex("689");
+    record.extractFirstSubfield("689", 't');
+
+    std::vector<std::string> vector;
+    size_t count = record.extractAllSubfields("100", &vector, "a0");
+
+    count = record.extractSubfield("689", '0', &vector);
+
+    count = record.extractSubfields("689", "02", &vector);
+
+    std::vector<std::pair<size_t, size_t>> LOKs;
+    count = record.findAllLocalDataBlocks(&LOKs);
+
+    if (count > 0) {
+        std::vector <size_t> indizes;
+        count = record.findFieldsInLocalBlock("852", "?1", LOKs[0], &indizes);
+    }
+
+    record.filterTags({"LOK"});
+    count = record.findAllLocalDataBlocks(&LOKs);
+
+    Subfields subfields('x', 'y');
+    subfields.addSubfield('a', "Test");
+    record.insertField("TST", subfields.toString());
+    ssize_t index = record.getFieldIndex("TST");
+    record.extractFirstSubfield("TST", 'a');
+
+    Subfields newSubfields('a', 'b');
+    newSubfields.addSubfield('a', "FooBar");
+    record.updateField(index, newSubfields.toString());
+    record.extractFirstSubfield("TST", 'a');
+    record.getFieldIndex("TST");
+
+    record.deleteField(index);
+    record.getFieldIndex("TST");
+
+    for (size_t i = 0; i < INSERTED_FIELDS_COUNT; ++i) {
+        record.insertField("TST", subfields.toString());
+    }
+
+    return record;
+}
+
+void speedTest(const std::string &inputFile) {
```

Comment:
Variable names, including dummy argument names should be lowercase with underscores.